CONTRIBUTING.md

```diff
@@ -1,59 +1,604 @@
+  or the user value it delivers, rather than any implementation detail.
+
+    eg: `"changed SomeFunc in file.go to also do thing X"`
+    is not useful for folks stopping by to quickly see what new stuff they can do with
+    `flintlock`, save that for commit messages or the PR description.
+    The title `"(feat): Add support for XYZ"` conveys the intent concisely and clearly.
+
+1. **There are new tests for new code.** This is very much required.
+
+1. **There are new tests for old code!** This will earn you the title of "Best Loved
+  and Respected Contributor" :boom: :sunglasses: .
+
+1. **There are well-written commit messages** ([see below](#commit-message-formatting))
+  which will make future debugging fun. (Any commits of the variety `"fix stuff"`, `"does thing"`
+  or `"my god why"` much be squashed or rebased.)
+
+1. **Each of those well-written commits is compilable** and the tests run successfully at
+  that point.
+
+In general, we will merge a PR once a maintainer has reviewed and approved it.
+Trivial changes (e.g., corrections to spelling) may get waved through.
+For substantial changes, more people may become involved, and you might get asked to resubmit the PR or divide the
+changes into more than one PR.
+
+### Commit message formatting
+
+_For more on how to write great commit messages, and why you should, check out
+[this excellent blog post](https://chris.beams.io/posts/git-commit/)._
+
+We follow a rough convention for commit messages that is designed to answer three
+questions: what changed, why was the change made, and how did you make it.
+
+The subject line should feature the _what_ and
+the body of the commit should describe the _why_ and _how_.
+If you encountered any weirdness along the way, this is a good place
+to note what you discovered and how you solved it.
+
+An example of a stellar commit message can be found [here](https://github.com/weaveworks/flintlock/commit/7a30dd99dc7c05827ba11050505c476799bb2932).
+
+The format can be described more formally as follows:
+
+```text
+<short title for what changed>
+<BLANK LINE>
+<why this change was made and what changed>
+<BLANK LINE>
+<any interesting details>
+<footer>
+```
+
+The first line is the subject and should be no longer than 70 characters, the
+second line is always blank, and other lines should be wrapped at a max of 80 characters.
+This allows the message to be easier to read on GitHub as well as in various git tools.
+
+There is a template recommend for use [here](https://gist.github.com/yitsushi/656e68c7db141743e81b7dcd23362f1a).
+
+# How the Maintainers process contributions
+
+## Prioritizing issues
+
+The core team regularly processes incoming issues. There may be some delay over holiday periods.
+
+Every issue will be assigned a `priority/<x>` label. The levels of priorities are:
+
+- [`critical`](https://github.com/weaveworks/flintlock/labels/priority%2Fcritical): These are time sensitive issues which should be picked up immediately.
```

Comment:
This label is `priority/critical-urgent`

Reply:
👍  ty